packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -706,6 +706,13 @@ export abstract class MappingTemplate {
         return this.fromString(`{"version": "2017-02-28", "operation": "Invoke", "payload": ${payload}}`);
     }
 
+    /**
+     * Mapping template to invoke a Lambda function with all available context fields
+     */
+    public static lambdaFullRequest(): MappingTemplate {
+        return this.fromString('{"version": "2017-02-28", "operation": "Invoke", "payload": $util.toJson($ctx)}');
+    }
```

Comment:
I think it's better to have just one of these functions that accepts something similar to an "AttributeValues" object (see https://github.com/aws/aws-cdk/pull/6236).

Reply:
@duarten I'm with you here. It looks like right now a user could get the same result of this function by calling the existing `lambdaRequest("$util.toJson($ctx)")` like so. It feels like we should be defining some generic data structure with helper methods to expose common mappings.

@hoegertn How do you feel about this?

Reply:
I agree that I could add it to the `lambdaRequest` method if no payload is given. But for further additions, I like having multiple methods for different use cases instead of option structures.

Reply:
I think a base level method with a standard structure and higher level methods that call that is what I see as the ideal. Then we aren't repeating logic in all the method bodies.

@shivlaks 🔔 just flagging this for your eyes/opnions.